backend/de.metas.adempiere.adempiere/base/src/main/java-legacy/org/compiere/util/TimeUtil.java

```diff
@@ -2110,4 +2110,24 @@ public static long getMillisBetween(@NonNull final Timestamp timestamp1, @NonNul
 	{
 		return timestamp2.getTime() - timestamp1.getTime();
 	}
+
+	public static boolean isOverlapping(@NonNull final Timestamp newValidFrom, @Nullable final Timestamp newValidTo, @NonNull final Timestamp start2, @Nullable final Timestamp end2)
```

Comment:
start1, end1

Reply:
TODO: junit tests for cases like:
1. 1---1    2--------------2
2. 1---12-------------2
3. 1---2--1-------------2
4. 2---1--1-----------2
5. 2-----------1---2----1
6. 2-----------21----1
7. 2-----------2       1----1

and add tests for null cases!!!!

Reply:
also assert 
* start1<=end1 , if not null
* start2 <= end2, if not null

Reply:
done